Two `@Incoming` via smallrye-reactive-messaging causes build errors.

## Describe the bug

I'm trying this experimental feature: https://smallrye.io/smallrye-reactive-messaging/smallrye-reactive-messaging/2/advanced/incomings.html using the `smallrye-kafka` connector.

I consistently get this failure when building the application:

```
[ERROR] Failed to execute goal io.quarkus:quarkus-maven-plugin:2.1.3.Final:build (default) on project smallrye-kafka-multiple-incomings: Failed to build quarkus application: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
[ERROR] 	[error]: Build step io.quarkus.smallrye.reactivemessaging.deployment.SmallRyeReactiveMessagingProcessor#build threw an exception: javax.enterprise.inject.spi.DefinitionException: SRMSG00052: Invalid method annotated with @Outgoing: org.acme.kafka.processor.Example#process - the method must not be `void`
[ERROR] 	at io.smallrye.reactive.messaging.MediatorConfigurationSupport.validatePublisher(MediatorConfigurationSupport.java:229)
[ERROR] 	at io.smallrye.reactive.messaging.MediatorConfigurationSupport.validate(MediatorConfigurationSupport.java:87)
[ERROR] 	at io.quarkus.smallrye.reactivemessaging.deployment.QuarkusMediatorConfigurationUtil.create(QuarkusMediatorConfigurationUtil.java:105)
[ERROR] 	at io.quarkus.smallrye.reactivemessaging.deployment.SmallRyeReactiveMessagingProcessor.build(SmallRyeReactiveMessagingProcessor.java:393)
[ERROR] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[ERROR] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[ERROR] 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[ERROR] 	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
[ERROR] 	at io.quarkus.deployment.ExtensionLoader$2.execute(ExtensionLoader.java:820)
[ERROR] 	at io.quarkus.builder.BuildContext.run(BuildContext.java:277)
[ERROR] 	at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
[ERROR] 	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2449)
[ERROR] 	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1478)
[ERROR] 	at java.base/java.lang.Thread.run(Thread.java:834)
[ERROR] 	at org.jboss.threads.JBossThread.run(JBossThread.java:501)
```

### Expected behavior

It should work [as described in the documentation](https://smallrye.io/smallrye-reactive-messaging/smallrye-reactive-messaging/2/advanced/incomings.html): it simply processes both messages from the channels in no particular order

### Actual behavior

There's a build failure that shows the usage of `@Outcoming`. I suspect the building of two `@Incoming` on the same function internally uses an `@Outcoming` to facilitate this, but the usage of `@Outcoming` makes the build step fail with `Invalid method annotated with @Outgoing: org.acme.kafka.processor.Example#process - the method must not be void`

## How to Reproduce?

A reproducer is here: https://github.com/navkast/quarkus-reproducer. It's a very simple reproducer amounting to just:
```java
  @Incoming("topic1")
  @Incoming("topic2")
  public CompletionStage<Void> process(Message<String> msg) {
    return msg.ack();
  }
```

Adding `@Blocking` or using `public void process(String msg) {` have the same results.

### Output of `uname -a` or `ver`

Darwin Kernel Version 20.6.0: Wed Jun 23 00:26:31 PDT 2021; root:xnu-7195.141.2~5/RELEASE_X86_64 x86_64

### Output of `java -version`

openjdk version "11.0.10" 2021-01-19 OpenJDK Runtime Environment (build 11.0.10+9) OpenJDK 64-Bit Server VM (build 11.0.10+9, mixed mode)

### Quarkus version or git rev

2.1.3.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.8.1 (05c21c65bdfed0f71a2f2ada8b84da59348c4c5d)

--- a/extensions/smallrye-reactive-messaging/deployment/src/main/java/io/quarkus/smallrye/reactivemessaging/deployment/QuarkusMediatorConfigurationUtil.java
+++ b/extensions/smallrye-reactive-messaging/deployment/src/main/java/io/quarkus/smallrye/reactivemessaging/deployment/QuarkusMediatorConfigurationUtil.java
@@ -5,12 +5,15 @@ import static io.quarkus.smallrye.reactivemessaging.deployment.ReactiveMessaging
 import static io.quarkus.smallrye.reactivemessaging.deployment.ReactiveMessagingDotNames.BROADCAST;
 import static io.quarkus.smallrye.reactivemessaging.deployment.ReactiveMessagingDotNames.COMPLETION_STAGE;
 import static io.quarkus.smallrye.reactivemessaging.deployment.ReactiveMessagingDotNames.INCOMING;
+import static io.quarkus.smallrye.reactivemessaging.deployment.ReactiveMessagingDotNames.INCOMINGS;
 import static io.quarkus.smallrye.reactivemessaging.deployment.ReactiveMessagingDotNames.KOTLIN_UNIT;
 import static io.quarkus.smallrye.reactivemessaging.deployment.ReactiveMessagingDotNames.MERGE;
 import static io.quarkus.smallrye.reactivemessaging.deployment.ReactiveMessagingDotNames.OUTGOING;
 import static io.quarkus.smallrye.reactivemessaging.deployment.ReactiveMessagingDotNames.SMALLRYE_BLOCKING;
 import static io.quarkus.smallrye.reactivemessaging.deployment.ReactiveMessagingDotNames.VOID_CLASS;
 
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.List;
 import java.util.concurrent.CompletionStage;
 import java.util.function.Supplier;
@@ -84,8 +87,11 @@ public final class QuarkusMediatorConfigurationUtil {
         }
         configuration.setParameterTypes(parameterTypes);
 
-        List<String> incomingValues = getValues(methodInfo, INCOMING);
+        // We need to extract the value of @Incoming and @Incomings (which contains an array of @Incoming)
+        List<String> incomingValues = new ArrayList<>(getValues(methodInfo, INCOMING));
+        incomingValues.addAll(getIncomingValues(methodInfo));
         configuration.setIncomings(incomingValues);
+
         String outgoingValue = getValue(methodInfo, OUTGOING);
         configuration.setOutgoing(outgoingValue);
 
@@ -244,6 +250,13 @@ public final class QuarkusMediatorConfigurationUtil {
     private static List<String> getValues(MethodInfo methodInfo, DotName dotName) {
         return methodInfo.annotations().stream().filter(ai -> ai.name().equals(dotName))
                 .map(ai -> ai.value().asString())
+                .collect(Collectors.toList());
+    }
+
+    private static List<String> getIncomingValues(MethodInfo methodInfo) {
+        return methodInfo.annotations().stream().filter(ai -> ai.name().equals(INCOMINGS))
+                .flatMap(incomings -> Arrays.stream(incomings.value().asNestedArray()))
+                .map(incoming -> incoming.value().asString())
                 .collect(Collectors.toList());
     }
 

--- a/extensions/smallrye-reactive-messaging/deployment/src/test/java/io/quarkus/smallrye/reactivemessaging/signatures/IncomingsTest.java
+++ b/extensions/smallrye-reactive-messaging/deployment/src/test/java/io/quarkus/smallrye/reactivemessaging/signatures/IncomingsTest.java
@@ -1,0 +1,76 @@
+package io.quarkus.smallrye.reactivemessaging.signatures;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.awaitility.Awaitility.await;
+
+import java.util.List;
+import java.util.concurrent.CopyOnWriteArrayList;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+
+import org.eclipse.microprofile.reactive.messaging.Incoming;
+import org.eclipse.microprofile.reactive.messaging.Outgoing;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+import org.reactivestreams.Publisher;
+
+import io.quarkus.test.QuarkusUnitTest;
+import io.smallrye.mutiny.Multi;
+
+public class IncomingsTest {
+
+    @RegisterExtension
+    static final QuarkusUnitTest config = new QuarkusUnitTest()
+            .setArchiveProducer(() -> ShrinkWrap.create(JavaArchive.class)
+                    .addClasses(ProducerOnA.class, ProducerOnB.class, MyBeanUsingMultipleIncomings.class));
+
+    @Inject
+    MyBeanUsingMultipleIncomings bean;
+
+    @Test
+    public void testIncomingsWithTwoSources() {
+        await().until(() -> bean.list().size() == 6);
+        assertThat(bean.list()).containsSubsequence("a", "b", "c");
+        assertThat(bean.list()).containsSubsequence("d", "e", "f");
+    }
+
+    @ApplicationScoped
+    public static class ProducerOnA {
+
+        @Outgoing("a")
+        public Publisher<String> produce() {
+            return Multi.createFrom().items("a", "b", "c");
+        }
+
+    }
+
+    @ApplicationScoped
+    public static class ProducerOnB {
+
+        @Outgoing("b")
+        public Publisher<String> produce() {
+            return Multi.createFrom().items("d", "e", "f");
+        }
+
+    }
+
+    @ApplicationScoped
+    public static class MyBeanUsingMultipleIncomings {
+
+        private final List<String> list = new CopyOnWriteArrayList<>();
+
+        @Incoming("a")
+        @Incoming("b")
+        public void consume(String s) {
+            list.add(s);
+        }
+
+        public List<String> list() {
+            return list;
+        }
+
+    }
+}

--- a/extensions/smallrye-reactive-messaging/runtime/src/main/java/io/quarkus/smallrye/reactivemessaging/runtime/QuarkusMediatorConfiguration.java
+++ b/extensions/smallrye-reactive-messaging/runtime/src/main/java/io/quarkus/smallrye/reactivemessaging/runtime/QuarkusMediatorConfiguration.java
@@ -212,7 +212,7 @@ public class QuarkusMediatorConfiguration implements MediatorConfiguration {
 
     @Override
     public String methodAsString() {
-        if (Arc.container() != null) {
+        if (Arc.container() != null && getBean() != null) {
             return getBean().getBeanClass().getName() + "#" + getMethodName();
         } else {
             return getMethodName();

